src/test/java/seedu/address/logic/commands/ResetCommandTest.java

```diff
@@ -0,0 +1,167 @@
+
+    @Test
+    public void constructor_invalidIndex_throwsParseException() {
+        assertThrows(ParseException.class, () ->
+                new ResetCommand(ParserUtil.parseIndex("-1"),
+                        new Tutorial(TUTORIAL_TO_TEST)).execute(markedModel));
+    }
+
+    /**
+     * Reset a person's attendance which is already not taken place.
+     */
+    @Test
+    public void execute_tutorialAlreadyReset_failure() {
+        ResetCommand resetCommand = new ResetCommand(INDEX_FIRST_PERSON, new Tutorial(TUTORIAL_TO_TEST));
+        try {
+            resetCommand.execute(markedModel);
+
+            // Second execution should fail
+            assertCommandFailure(resetCommand, markedModel, String.format(Messages.MESSAGE_RESET_UNNECESSARY,
+                    Messages.format(markedModel.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased())),
+                    new Tutorial("1").tutorial));
+        } catch (CommandException e) {
+            // Should not get here
+            throw new RuntimeException();
+        }
+    }
+
+    @Test
+    public void execute_success() {
+        Tutorial tutorialToBeAdded = new Tutorial(TUTORIAL_TO_TEST);
+
+        Person personToEdit = markedModel.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        ResetCommand resetCommand = new ResetCommand(INDEX_FIRST_PERSON, tutorialToBeAdded);
+
+        try {
+            resetCommand.execute(markedModel);
+        } catch (CommandException e) {
+            throw new RuntimeException(e);
+        }
+
+        Person editedPerson = markedModel.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+
+        // Check person is edited
+        Map<Tutorial, AttendanceStatus> newTutorials = new HashMap<>(personToEdit.getTutorials());
+        newTutorials.put(tutorialToBeAdded, AttendanceStatus.NOT_TAKEN_PLACE);
+        Person expectedEditedPerson = new Person(
+                personToEdit.getName(),
+                personToEdit.getStudentId(),
+                personToEdit.getPhone(),
+                personToEdit.getEmail(),
+                personToEdit.getTags(),
+                newTutorials
+        );
+        assertEquals(expectedEditedPerson, editedPerson);
+
+        // Check model is updated with new person attribute
+        Model expectedModel = new ModelManager(new AddressBook(markedModel.getAddressBook()), new UserPrefs());
+        String expectedMessage = String.format(Messages.MESSAGE_RESET_SUCCESS, Messages.format(editedPerson),
+                tutorialToBeAdded.tutorial);
+        Model typicalModel = new ModelManager(getMarkedAddressBook(TUTORIAL_TO_TEST), new UserPrefs());
+        assertCommandSuccess(resetCommand, typicalModel, expectedMessage, expectedModel);
+    }
+
+    /**
+     * Reset all persons in the list.
+     */
+    @Test
+    public void execute_shouldResetAll_success() {
+        Tutorial tutorialToBeAdded = new Tutorial(TUTORIAL_TO_TEST);
```

Comment:
tutorialToBeAdded seems like a misleading name?

Reply:
Resolved, amended to tutorialToBeReset